packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts

```diff
@@ -264,6 +264,15 @@ export function frameSnapshotStreamer(snapshotStreamer: string) {
       }
     }
 
+    private _isInsideHeadTag(e: Element): boolean {
```

Comment:
I wonder if you'd like an approach like we take with `shadowDomNesting` better, since we are already traversing the whole DOM.

Reply:
Done. Not sure if I like it better.